lib/models/instrumentation.js

```diff
@@ -0,0 +1,277 @@
+      if (nextStat.done) { break; }
+
+      statName = nextStat.value[0];
+      statValue = nextStat.value[1];
+
+      if (statName === 'time.self') {
+        totalTime += statValue;
+      }
+    }
+  }
+
+  var summary = {
+    build: {
+      type: resultAnnotation.type,
+      count: this.instrumentations.build.count,
+      outputChangedFiles: result.outputChanges,
+    },
+    output: result.directory,
+    totalTime: totalTime,
+    buildSteps: buildSteps,
+  };
+
+  if (resultAnnotation.type === 'rebuild') {
+    summary.build.primaryFile = resultAnnotation.primaryFile;
+    summary.build.changedFileCount = resultAnnotation.changedFiles.length;
+    summary.build.changedFiles = resultAnnotation.changedFiles.slice(0, 10);
+  }
+
+  return summary;
+};
+
+function totalTime(tree) {
+  var totalTime = 0;
+  var nodeItr;
+  var node;
+  var statName;
+  var statValue;
+  var statsItr;
+  var nextNode;
+  var nextStat;
+
+  for (nodeItr = tree.preOrderIterator();;) {
+    nextNode = nodeItr.next();
+    if (nextNode.done) { break; }
+
+    node = nextNode.value;
+
+    for (statsItr = node.statsIterator();;) {
+      nextStat = statsItr.next();
+      if (nextStat.done) { break; }
+
+      statName = nextStat.value[0];
+      statValue = nextStat.value[1];
+
+      if (statName === 'time.self') {
+        totalTime += statValue;
+      }
+    }
+  }
+
+  return totalTime;
+}
+
+Instrumentation.prototype._initSummary = function (tree) {
+  return {
+    totalTime: totalTime(tree)
+  };
+};
+
+Instrumentation.prototype._commandSummary = function (tree, commandName, commandArgs) {
+  return {
+    name: commandName,
+    args: commandArgs,
+    totalTime: totalTime(tree)
+  };
+};
+
+Instrumentation.prototype._shutdownSummary = function (tree) {
+  return {
+    totalTime: totalTime(tree)
+  };
+};
+
+Instrumentation.prototype._instrumentationFor = function (name) {
+  var instr = this.instrumentations[name];
+  if (!instr) {
+    throw new Error('No such instrumentation "' + name + '"');
+  }
+  return instr;
+};
+
+Instrumentation.prototype._instrumentationTreeFor = function(name) {
+  return tree.loadFromNode(this.instrumentations[name].node);
+};
+
+Instrumentation.prototype._invokeAddonHook = function (name, instrumentationInfo) {
+  if (!experiments.INSTRUMENTATION) { return; }
+
+  if (this.project && this.project.addons.length) {
+    this.project.addons.forEach(function (addon) {
+      var hook = addon[experiments.INSTRUMENTATION];
+      if (typeof hook === 'function') {
+        hook.call(addon, name, instrumentationInfo);
+      } else if (name === 'build') {
+        hook = addon[experiments.BUILD_INSTRUMENTATION];
+        if (typeof hook === 'function') {
+          hook.call(addon, instrumentationInfo);
+        }
+      }
+    });
+  }
+};
+
+Instrumentation.prototype._writeInstrumentation = function (name, instrumentationInfo) {
+  if (!vizEnabled()) { return; }
+
+  var filename = 'broccoli-viz.' + name;
```

Comment:
> I think broccoli-viz is generic enough to handle any of them, although it's certainly built for build.

Broccoli is very specifically a build tool, and it is **only** related to the `build` instrumentation. Using the term "broccoli" in relation to ember-cli init/ commands / shutdown. 

> instrumentation.build.0.json?

Yes, I like this.